astropy/wcs/tests/test_utils.py

```diff
@@ -353,6 +368,52 @@ def test_wcs_to_celestial_frame():
     assert isinstance(frame, Galactic)
 
 
+def test_wcs_to_body_frame():
+    # Import astropy.coordinates here to avoid circular imports
+    from astropy.coordinates import BaseCoordinateFrame
+
+    unknown_wcs = WCS(naxis=2)
+    unknown_wcs.wcs.ctype = ["UTLN-TAN", "UTLT-TAN"]
+    with pytest.raises(KeyError, match="unknown solar system object abbreviation UT"):
+        frame = wcs_to_celestial_frame(unknown_wcs)
+
+    triaxial_wcs = WCS(naxis=2)
+    triaxial_wcs.wcs.ctype = ["MELN-TAN", "MELT-TAN"]
+    triaxial_wcs.wcs.aux.a_radius = 2439700.0
+    triaxial_wcs.wcs.aux.b_radius = 2439900.0
+    triaxial_wcs.wcs.aux.c_radius = 2438800.0
+    with pytest.raises(
+        NotImplementedError, match="triaxial systems are not supported at this time"
+    ):
+        frame = wcs_to_celestial_frame(triaxial_wcs)
+
+    mywcs = WCS(naxis=2)
+    mywcs.wcs.ctype = ["MALN-TAN", "MALT-TAN"]
+    mywcs.wcs.dateobs = "2017-08-17T12:41:04.430"
+    mywcs.wcs.name = "Mars Bodycentric Body-Fixed"
+
+    mywcs.wcs.aux.a_radius = 3396190.0
+    mywcs.wcs.aux.b_radius = 3396190.0
+    mywcs.wcs.aux.c_radius = 3376190.0
+    frame = wcs_to_celestial_frame(mywcs)
+    assert issubclass(frame, BaseCoordinateFrame)
+    assert issubclass(frame.representation_type, BaseBodycentricRepresentation)
+    assert frame.name == "Mars"
+    assert frame.representation_type._equatorial_radius == 3396190.0 * u.m
```

Comment:
Also test `_flattening == 0.0`?

Reply:
I have tested Mars flattening and added a case where flattening == 0 (Venus)